juju/juju
# Manual Provider Subnet Discovery

## Description of change

This change allows us to turn on subnet discovery during update of link-layer devices detected by machine agents.

If the model is for a _manual_ cloud, incoming link-layer device data causes subnets that we have not yet observed to be added to state.

A following patch will turn on provider space support and ensure that constraints/bindings can be honoured in manual clouds.

## QA steps

Assuming access to a machine with IP 172.16.99.2:
- `juju bootstrap manual/ubuntu@172.16.99.2 manual-subnets --no-gui --debug`.
- `juju switch controller`.
- `juju subnets` should show:
```yaml
subnets:
  172.16.99.0/24:
    type: ipv4
    status: in-use
    space: alpha
    zones: []
```

## Documentation changes

Once space support is turned on, I will ensure appropriate documentation changes.

## Bug reference

N/A
